Dockerfile

```diff
@@ -12,7 +12,7 @@
 # the License.
 #
 
-FROM openjdk:alpine
+FROM openjdk:8u232-jdk-slim-buster
```

Comment:
This will get equally outdated if we pin an exact version.

I would rather use tag `8-jdk-alpine` it is also recommended by readme https://hub.docker.com/_/openjdk?tab=description&page=3&name=alpine `<version>-alpine`. Would that work for you? I have pulled the image and it uses `1.8.0_212`. I think `openjdk:alpine` is not maintained anymore.


https://java.com/en/download/faq/release_dates.xml

Reply:
Thank you the review and comments! please see below:

> This will get equally outdated if we pin an exact version

True, the above merely pins the dependency. This ensures a repeatable builds, makes the base version explicit among other properties. For example see https://snyk.io/blog/10-docker-image-security-best-practices/ item 6.

> I would rather use tag 8-jdk-alpine it is also recommended by readme https://hub.docker.com/_/openjdk?tab=description&page=3&name=alpine <version>-alpine. Would that work for you? I have pulled the image and it uses 1.8.0_212. I think openjdk:alpine is not maintained anymore.

The maintainability issue for alpine and JDK 8 is one of the reasons for using debian base. It shouldn't be a problem for JDK 11 though. AFAIK Alpine build depend on IcedTea which has a slower release cycle.

The 8u232-jre-slim-buster is 67.53 compressed. 8u212-jre-alpine3.9 is 55.02. Not that much of a difference when the main jar is at ~100MB?

I hope to make a second PR to refactor the dockerfile to be multi-stage and use a JRE for the final image.

Reply:
The image size I see is:
```
openjdk                                                        8u232-jdk-slim-buster   971671e78456        3 days ago          284MB
```

Again I would prefer not pinning a specific JDK version. There is no rebuilding process in place for this upstream image. If you are concerned with security fixes quickly reaching this image consider maintaining your own image.

Reply:
> Would pinning to 8u212-jdk-alpine3.9 suffice?

No, I would prefer to have only 8 in the name and not the patch version.

Reply:
Unfortunately it seems we are unable to reach an agreement. With the above there is no way to ensure minimal build > 8b182 is used.

My apologies for wasting your time. I guess maintaining a fork is the on;y way forward for us.

Reply:
You could use `openjdk:8-jdk-slim` which uses `1.8.0_232`.

Reply:
Here it is https://github.com/jaegertracing/spark-dependencies/pull/81

Reply:
@pavolloffay thanks for clearing things up! again my apologies for the misunderstanding :)